### PR TITLE
Added method to get list of claiming domains for a given wallet ID

### PR DESCRIPTION
**Pull request by MattCharles**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -293,17 +297,37 @@ export const createClaimWithChildInstance = (
...
         signer
       );
       return tx;
     },
+    getClaimingIDsForUser: async (
+      walletID: string,
+      signer: ethers.Signer
+    ): Promise<string[]> => {
+      const claimingToken = await getClaimingToken(
+        signer,
```
Comment (Remscar) [inline src/index.ts:313]: use the provider, not a signer, since this is only being used for view methods

File: src/contracts/index.ts [hunk 2]
```diff
@@ -21,3 +23,14 @@ export const getClaimContract = async (
   const contract = ClaimWithChildSale__factory.connect(address, provider);
   return contract;
 };
+
+export const getClaimRegistrar = async (
+  provider: ethers.providers.Provider | ethers.Signer,
```
Comment (RyanCross) [inline src/contracts/index.ts:28]: nit: remove signer so that consumers are forced to use a provider

File: src/index.ts [hunk 3]
```diff
@@ -293,17 +297,37 @@ export const createClaimWithChildInstance = (
...
+        config.claimingRegistrarAddress
+      );
+      let childDomain: ethers.BigNumber;
+      const claimingIDs: string[] = [];
+      const bigNumOwned = await claimingToken.balanceOf(walletID);
```
Comment (RyanCross) [inline src/index.ts:318]: nit: `bigNumDomainsOwned`?

File: src/index.ts [hunk 4]
```diff
@@ -293,17 +297,37 @@ export const createClaimWithChildInstance = (
...
+      const numOwned = bigNumOwned.toNumber();
```
Comment (RyanCross) [inline src/index.ts:319]: `numDomainsOwned`?

**Review by RyanCross — APPROVED** [review]

**Review by Remscar — APPROVED** [review]

File: src/index.ts [hunk 5]
```diff
@@ -293,17 +297,34 @@ export const createClaimWithChildInstance = (
...
+        config.web3Provider,
+        config.claimingRegistrarAddress
+      );
+      let childDomain: ethers.BigNumber;
+      const claimingIDs: string[] = [];
+      const bigNumDomainsOwned = await claimingToken.balanceOf(walletID);
+      const numDomainsOwned = bigNumDomainsOwned.toNumber();
+      if (numDomainsOwned != 0) {
+        let i = 0;
+        while (i < numDomainsOwned) {
```
Comment (JamesEarle) [inline src/index.ts:319]: You don't increment the value of `i` here. Maybe you meant `while (claimingIDs.length < numDomainsOwned)` ? If so delete `let i = 0` and update this condition

**Review by JamesEarle — APPROVED** [review]
looks good one comment

File: src/index.ts (hunk 5 above)
Comment (MattCharles) [reply src/index.ts:319]: great catch

**Pull request merged**